sdk/storage/Azure.Storage.Files.DataLake/README.md

```diff
@@ -33,39 +34,178 @@ az storage account create --name MyStorageAccount --resource-group MyResourceGro
 
 ## Key concepts
 
-TODO: Add Key Concepts
+This preview package for .NET includes ADLS Gen2 specific API support made available in Blob SDK. This includes:
+1. New directory level operations (Create, Rename/Move, Delete) for hierarchical namespace enabled (HNS) storage accounts. For HNS enabled accounts, the rename/move operations are atomic.
+2. Permission related operations (Get/Set ACLs) for hierarchical namespace enabled (HNS) accounts. 
+
+HNS enabled accounts in ADLS Gen2 can also now leverage most of the operations available in Blob SDK. Support for File level semantics for ADLS Gen2 is planned to be made available in Blob SDK in a later release. In the meantime, please find below mapping for ADLS Gen2 terminology to Blob terminology.
+
+|ADLS Gen2 	                | Blob       |
+| --------------------------| ---------- |
+|Filesystem                 | Container  | 
+|Path (File or Directory)   | Blob       |
+
+Note: This SDK does not support hierarchical namespace (HNS) disabled storage accounts.
 
 ## Examples
 
-TODO: Add Examples
+### Create a DataLakeServiceClient
+```C# Snippet:SampleSnippetDataLakeServiceClient_Create
```

Comment:
Do you have corresponding `#region`s?  I don't see them in the PR and it's breaking the CI builds.

Reply:
What do you mean by regions?

Reply:
You need a `#region` in your C# samples named `Snippet:SampleSnippetDataLakeServiceClient_Create` and the code will get pulled in when you run `C:\src\azure-sdk-for-net\eng\Update-Snippets.ps1`.

See https://github.com/Azure/azure-sdk-for-net/pull/7928 for an overview and the [README](https://raw.githubusercontent.com/Azure/azure-sdk-for-net/master/sdk/storage/Azure.Storage.Queues/README.md)/[Samples](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Queues/samples/Sample01a_HelloWorld.cs#L31) for Queues.  It's helping us ensure our README code snippets are always up to date and compiling.